src/gatsby/models/authors/parse-link.js

```diff
@@ -1,31 +1,77 @@
-const processStringLink = require('./string-link-processor.js')
+/*
+   Processor builder: makes a function that turns a string into processed Link metadata
+
+   Takes "Mod functions" as parameters, returns a function that starts with {} and pipes
+   that through each function, then returns the result.
+
+   signature: ({ scheme: string, host: string, pathname: string}, input)
+   example:      ^ https://      ^ mysite.com  ^ /rest/of/the/url ^ an object
+
+   The second parameter will be the current state of the input, and usually will
+   have to be spread into your return result.
+ */
+function processor(...mods) {
+  return function process(groups) {
+    return mods.reduce(
+      (currentResult, _, i) => mods[i](groups, currentResult),
+      {}
+    )
+  }
+}
+
+// Processor helpers
+
+function trimSlashes(input) {
+  return /^\/?(.*?)\/?$/.exec(input)[1]
+}
+
+const asSite = site => (groups, input) => ({
+  ...input,
+  site
```

Comment:
should it be "site: site"?

Reply:
This is ES5 syntactic sugar for the same thing.

Reply:
👍